crates/forge/tests/integration/mock_call.rs

```diff
@@ -0,0 +1,424 @@
+use crate::integration::common::corelib::{corelib, predeployed_contracts};
+use crate::integration::common::runner::Contract;
+use crate::{assert_passed, test_case};
+use camino::Utf8PathBuf;
+use forge::run;
+use indoc::indoc;
+use std::path::Path;
+
+#[test]
+fn start_mock_call_simple() {
+    let test = test_case!(
+        indoc!(
+            r#"
+        use result::ResultTrait;
+        use cheatcodes::PreparedContract;
+        use array::ArrayTrait;
+
+        #[starknet::interface]
+        trait IMockChecker<TContractState> {
+            fn get_thing(self: @TContractState) -> felt252;
+        }
+
+        #[test]
+        fn start_mock_call_simple() {
+            let mut calldata = ArrayTrait::new();
+            calldata.append(420);
+
+            let class_hash = declare('MockChecker');
+            let prepared = PreparedContract { class_hash: class_hash, constructor_calldata: @calldata };
+            let contract_address = deploy(prepared).unwrap();
+
+            let dispatcher = IMockCheckerDispatcher { contract_address };
+
+            let mut mock_ret_data = ArrayTrait::new();
+            mock_ret_data.append(421);
+            start_mock_call(contract_address, 'get_thing', mock_ret_data);
+
+            let thing = dispatcher.get_thing();
+
+            assert(thing == 421, 'Incorrect thing');
+        }
+
+        #[test]
+        fn start_mock_call_simple_mock_before_dispatcher_created() {
+            let mut calldata = ArrayTrait::new();
+            calldata.append(420);
+
+            let class_hash = declare('MockChecker');
+            let prepared = PreparedContract { class_hash: class_hash, constructor_calldata: @calldata };
+            let contract_address = deploy(prepared).unwrap();
+
+            let mut mock_ret_data = ArrayTrait::new();
+            mock_ret_data.append(421);
+            start_mock_call(contract_address, 'get_thing', mock_ret_data);
+
+            let dispatcher = IMockCheckerDispatcher { contract_address };
+            let thing = dispatcher.get_thing();
+
+            assert(thing == 421, 'Incorrect thing');
+        }
+    "#
+        ),
+        Contract::from_code_path(
+            "MockChecker".to_string(),
+            Path::new("tests/data/contracts/mock_checker.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run(
+        &test.path().unwrap(),
+        &test.path().unwrap().join("src/lib.cairo"),
+        &Some(test.linked_libraries()),
+        &Default::default(),
+        Some(&Utf8PathBuf::from_path_buf(corelib().to_path_buf()).unwrap()),
+        &test.contracts(corelib().path()).unwrap(),
+        &Utf8PathBuf::from_path_buf(predeployed_contracts().to_path_buf()).unwrap(),
+    )
+    .unwrap();
```

Comment:
I think we could create some helper function for this one because it is used in every test

Reply:
Can you create an issue for that?

Reply:
added https://github.com/foundry-rs/starknet-foundry/issues/432